aicsimageio/readers/default_reader.py

```diff
@@ -1,89 +1,438 @@
+                with imageio.get_reader(open_resource, format=extension, mode=mode):
+                    return True
+
+        # Exceptions that are raised by imageio for unsupported file types
+        except (ValueError, IndexError):
+            return False
+
+        # Some FFMPEG formats and reading just suck
+        # If they can't get metadata remotely they throw an OSError because ffmpeg is
+        # ran through subprocess (I believe)
+        # If we let the stack trace go, user would receive:
+        #
+        # OSError: Could not load meta information
+        # === stderr ===
+        #
+        # ffmpeg version 4.2.2-static https://johnvansickle.com/ffmpeg/
+        # Copyright (c) # 2000-2019 the FFmpeg developers
+        # ...
+        # /tmp/imageio_cbof2u37: Invalid data found when processing input
+        except OSError:
+            raise IOError(REMOTE_READ_FAIL_MESSAGE.format(path=path))
 
     def __init__(self, image: types.PathLike):
+        """
+        A catch all for image file reading that defaults to using imageio
+        implementations.
+
+        Parameters
+        ----------
+        image: types.PathLike
+            Path to image file to construct Reader for.
+        """
         # Expand details of provided image
-        self.abstract_file = io_utils.pathlike_to_fs(image, enforce_exists=True)
+        self.fs, self.path = io_utils.pathlike_to_fs(image, enforce_exists=True)
+        self.extension, self.imageio_read_mode = self._get_extension_and_mode(self.path)
 
+        # Enforce valid image
+        if not self._reader_supports_image(self.fs, self.path):
+            raise exceptions.UnsupportedFileFormatError(self.extension)
 
-FFMPEG_FORMATS = ["mov", "avi", "mpg", "mpeg", "mp4", "mkv", "wmv", "ogg"]
+    @staticmethod
+    def guess_dim_order(shape: Tuple[int]) -> str:
+        if len(shape) == 2:
+            return f"{DimensionNames.SpatialY}{DimensionNames.SpatialX}"
+        elif len(shape) == 3:
+            return (
+                f"{DimensionNames.SpatialY}{DimensionNames.SpatialX}"
+                f"{DimensionNames.Channel}"
+            )
+        elif len(shape) == 4:
+            return (
+                f"{DimensionNames.Time}{DimensionNames.SpatialY}"
+                f"{DimensionNames.SpatialX}{DimensionNames.Channel}"
+            )
 
-# BIG BUCK BUNNY 15 SECONDS:
-# "https://archive.org/embed/archive-video-files/test.mp4"
+        return Reader.guess_dim_order(shape)
 
-# AICS TEST RESOURCES
-# s3://aics-modeling-packages-test-resources/aicsimageio/test_resources/resources/example.png
-# s3://aics-modeling-packages-test-resources/aicsimageio/test_resources/resources/example.jpg
-# s3://aics-modeling-packages-test-resources/aicsimageio/test_resources/resources/example.gif
-# s3://aics-modeling-packages-test-resources/aicsimageio/test_resources/resources/s_1_t_1_c_1_z_1.tiff
+    @property
+    def scenes(self) -> Tuple[int]:
```

Comment:
Can you explain this? I'm guessing this reader won't encounter multiscene files?

Reply:
Yes that is the assumption for `DefaultReader`. Fortunately, if we do encounter a file type that default reader hits and wants scenes, we can just update these scene functions.